backend/src/judge/middleware/middleware.ts

```diff
@@ -0,0 +1,24 @@
+export interface CodeDetail {
+  functionName: string;
+  argTypes: string[];
+  variableNames: string[];
+  returnType: string;
+}
```

Comment:
Perhaps this will offer better safety to the correspondence between arguments and their type?

```suggestion
export interface CodeDetail {
  functionName: string;
  arguments: { type: string; name: string }[];
  returnType: string;
}
```

Reply:
Yep this looks better